pkg/network/protocols/kafka/api_version_counter.go

```diff
@@ -14,23 +14,24 @@ import (
 // APIVersionCounter is a Kafka API version aware counter, it has a counter for each supported Kafka API version.
 // It enables the use of a single metric that increments based on the API version, avoiding the need for separate metrics for API version
 type APIVersionCounter struct {
-	hitsV1, hitsV2, hitsV3, hitsV4, hitsV5, hitsV6, hitsV7, hitsV8, hitsV9, hitsV10, hitsV11 *libtelemetry.Counter
+	hitsV1, hitsV2, hitsV3, hitsV4, hitsV5, hitsV6, hitsV7, hitsV8, hitsV9, hitsV10, hitsV11, hitsUnsupportedVersion *libtelemetry.Counter
 }
```

Comment:
wondering if we can use array here instead of a massive duplication

```go
type APIVersionCounter struct {	type APIVersionCounter struct {
	hitsVersions []*libtelemetry.Counter
}
```

Reply:
initialization:
```go

for i := 0; i < maxSize; i++ {
    hitsVersions[i] = metricGroup.NewCounter(metricName, append(tags, fmt.Sprintf("protocol_version:%d", i+1)...)
}
```

Reply:
Good idea, changed the implementation